beets: 2.0.0 -> 2.2.0

astratagem opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

<https://github.com/beetbox/beets/blob/v2.1.0/docs/changelog.rst>

Note that this release includes a lot of improvements to the upstream project's build system, so a refactor of `common.nix` was in order.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

astratagem committed:
beets: alphabetize builtin plugins

Most were already in order but some plugins were thrown in at the bottom of the
file.

astratagem committed:
beets: mark builtin acousticbrainz plugin as deprecated

This happened in v2.0.0.

astratagem committed:
beets: add @montchr as co-maintainer

kleinops commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358086`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>beetsPackages.extrafiles</li>
    <li>beetsPackages.extrafiles.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>beets-unstable</li>
    <li>beets-unstable.dist</li>
    <li>beets-unstable.doc</li>
    <li>beets-unstable.man</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>beets</li>
    <li>beets.dist</li>
    <li>beets.doc</li>
    <li>beets.man</li>
  </ul>
</details>

montchr commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358086`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>beetsPackages.extrafiles</li>
    <li>beetsPackages.extrafiles.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>beets-unstable</li>
    <li>beets-unstable.dist</li>
    <li>beets-unstable.doc</li>
    <li>beets-unstable.man</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>beets</li>
    <li>beets.dist</li>
    <li>beets.doc</li>
    <li>beets.man</li>
  </ul>
</details>

astratagem committed:
beetsPackages.extrafiles: remove broken/unmaintained plugin

As noted in a comment in the package expression, the third-party `extrafiles`
plugin has not been maintained since 2020 and has been broken since the release
of beets version 2.0.0 in May 2024.

Since the `beets-stable` package has been updated to version 2.0.0, it is no
longer feasible to keep `beetsPackages.extrafiles` around.

montchr commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358086`

---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>beets</li>
    <li>beets-unstable</li>
    <li>beets-unstable.dist</li>
    <li>beets-unstable.doc</li>
    <li>beets-unstable.man</li>
    <li>beets.dist</li>
    <li>beets.doc</li>
    <li>beets.man</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
  </ul>
</details>

montchr commented:
All of the tests for `beetsPackages.copyartifacts` that weren't already being skipped now fail:

```
=========================== short test summary info ============================
FAILED tests/test_filename.py::CopyArtifactsFilename::test_import_dir_with_illegal_character_in_album_name - AttributeError: module 'beets.util' has no attribute 'py3>
FAILED tests/test_filename.py::CopyArtifactsFilename::test_import_dir_with_unicode_character_in_artifact_name_copy - AttributeError: module 'beets.util' has no attrib>
FAILED tests/test_filename.py::CopyArtifactsFilename::test_import_dir_with_unicode_character_in_artifact_name_move - AttributeError: module 'beets.util' has no attrib>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_copy_all_artifacts_by_default - AttributeError: module 'beets.util' has no attribute 'py3>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_copy_artifacts - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_do_nothing_when_not_copying_or_moving - AttributeError: module 'beets.util' has no attrib>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_ignore_media_files - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_move_artifacts - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_only_copy_artifacts_matching_configured_extension - AttributeError: module 'beets.util' h>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_prune_import_directory_when_emptied - AttributeError: module 'beets.util' has no attribut>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_rename_when_copying - AttributeError: module 'beets.util' has no attribute 'py3_path'. Di>
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_rename_when_moving - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did>
FAILED tests/test_printignored.py::CopyArtifactsPrintIgnoredTest::test_do_not_print_ignored_by_default - AttributeError: module 'beets.util' has no attribute 'py3_pat>
FAILED tests/test_printignored.py::CopyArtifactsPrintIgnoredTest::test_print_ignored - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_do_nothing_when_paths_do_not_change_with_copy_import - AttributeError: module 'beets.util' has no attri>
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_do_nothing_when_paths_do_not_change_with_move_import - AttributeError: module 'beets.util' has no attri>
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_prune_empty_directories_with_copy_import - AttributeError: module 'beets.util' has no attribute 'py3_pa>
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_reimport_artifacts_with_copy - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you>
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_reimport_artifacts_with_move - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you>
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_rename_when_paths_do_not_change - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did >
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_rename_with_copy_import - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_rename_with_move_import - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
SKIPPED [1] tests/test_reimport.py:152: Todo
SKIPPED [1] tests/test_reimport.py:77: Failing
================== 22 failed, 2 skipped, 3 warnings in 0.40s ===================
```

Unfortunately the upstream plugin repo hasn't seen a commit since 2023-01-07.  I didn't see anything that looked relevant in the beets project changelog, and I am not a Python person so might not come up with a sensible solution.  I'd appreciate input from others who have more experience in that domain!  I haven't tested the plugin manually, but based on those errors, it looks pretty clear that it won't work.  

Maybe `beetsPackages.copyartifacts` should be marked as broken for now?

<details>
<summary>Full build log for <code>beetsPackages.copyartifacts</code></summary>

```log
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/mmj691ykpqli7slkqkqmb2bw0bijqpnm-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build/lib/beetsplug
copying beetsplug/__init__.py -> build/lib/beetsplug
copying beetsplug/copyartifacts.py -> build/lib/beetsplug
/nix/store/yhx91ydywmrr32qhksc8fg2lxkp0s7kf-python3.12-setuptools-75.1.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
installing to build/bdist.linux-aarch64/wheel
running install
running install_lib
creating build/bdist.linux-aarch64/wheel
creating build/bdist.linux-aarch64/wheel/beetsplug
copying build/lib/beetsplug/__init__.py -> build/bdist.linux-aarch64/wheel/./beetsplug
copying build/lib/beetsplug/copyartifacts.py -> build/bdist.linux-aarch64/wheel/./beetsplug
running install_egg_info
running egg_info
creating beets_copyartifacts3.egg-info
writing beets_copyartifacts3.egg-info/PKG-INFO
writing dependency_links to beets_copyartifacts3.egg-info/dependency_links.txt
writing top-level names to beets_copyartifacts3.egg-info/top_level.txt
writing manifest file 'beets_copyartifacts3.egg-info/SOURCES.txt'
reading manifest file 'beets_copyartifacts3.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'beets_copyartifacts3.egg-info/SOURCES.txt'
Copying beets_copyartifacts3.egg-info to build/bdist.linux-aarch64/wheel/./beets_copyartifacts3-0.1.5-py3.12.egg-info
running install_scripts
creating build/bdist.linux-aarch64/wheel/beets_copyartifacts3-0.1.5.dist-info/WHEEL
creating 'dist/beets_copyartifacts3-0.1.5-py3-none-any.whl' and adding 'build/bdist.linux-aarch64/wheel' to it
adding 'beetsplug/__init__.py'
adding 'beetsplug/copyartifacts.py'
adding 'beets_copyartifacts3-0.1.5.dist-info/LICENSE'
adding 'beets_copyartifacts3-0.1.5.dist-info/METADATA'
adding 'beets_copyartifacts3-0.1.5.dist-info/WHEEL'
adding 'beets_copyartifacts3-0.1.5.dist-info/top_level.txt'
adding 'beets_copyartifacts3-0.1.5.dist-info/RECORD'
removing build/bdist.linux-aarch64/wheel
Finished executing setuptoolsBuildPhase
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed beets_copyartifacts3-0.1.5-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: giDiscoverSelf
@nix { "action": "setPhase", "phase": "giDiscoverSelf" }
Running phase: giDiscoverSelf
@nix { "action": "setPhase", "phase": "giDiscoverSelf" }
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/yl6znwxfxmp9cjkrhkndkxrfhr78axyp-beets-copyartifacts-0.1.5
checking for references to /build/ in /nix/store/yl6znwxfxmp9cjkrhkndkxrfhr78axyp-beets-copyartifacts-0.1.5...
patching script interpreter paths in /nix/store/yl6znwxfxmp9cjkrhkndkxrfhr78axyp-beets-copyartifacts-0.1.5
stripping (with command strip and flags -S -p) in  /nix/store/yl6znwxfxmp9cjkrhkndkxrfhr78axyp-beets-copyartifacts-0.1.5/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/141mzbj1siazs1gac0ks42v0vdjr5ijs-beets-copyartifacts-0.1.5-dist
checking for references to /build/ in /nix/store/141mzbj1siazs1gac0ks42v0vdjr5ijs-beets-copyartifacts-0.1.5-dist...
patching script interpreter paths in /nix/store/141mzbj1siazs1gac0ks42v0vdjr5ijs-beets-copyartifacts-0.1.5-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
rootdir: /build/source
collected 24 items                                                             

tests/test_filename.py FFF                                               [ 12%]
tests/test_flatdirectory.py FFFFFFFFF                                    [ 50%]
tests/test_printignored.py FF                                            [ 58%]
tests/test_reimport.py FFsFsFFFFF                                        [100%]

=================================== FAILURES ===================================
__ CopyArtifactsFilename.test_import_dir_with_illegal_character_in_album_name __

self = <tests.test_filename.CopyArtifactsFilename testMethod=test_import_dir_with_illegal_character_in_album_name>

    def setUp(self):
>       super(CopyArtifactsFilename, self).setUp()

tests/test_filename.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_filename.CopyArtifactsFilename testMethod=test_import_dir_with_illegal_character_in_album_name>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_ CopyArtifactsFilename.test_import_dir_with_unicode_character_in_artifact_name_copy _

self = <tests.test_filename.CopyArtifactsFilename testMethod=test_import_dir_with_unicode_character_in_artifact_name_copy>

    def setUp(self):
>       super(CopyArtifactsFilename, self).setUp()

tests/test_filename.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_filename.CopyArtifactsFilename testMethod=test_import_dir_with_unicode_character_in_artifact_name_copy>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_ CopyArtifactsFilename.test_import_dir_with_unicode_character_in_artifact_name_move _

self = <tests.test_filename.CopyArtifactsFilename testMethod=test_import_dir_with_unicode_character_in_artifact_name_move>

    def setUp(self):
>       super(CopyArtifactsFilename, self).setUp()

tests/test_filename.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_filename.CopyArtifactsFilename testMethod=test_import_dir_with_unicode_character_in_artifact_name_move>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
____ CopyArtifactsFromFlatDirectoryTest.test_copy_all_artifacts_by_default _____

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_copy_all_artifacts_by_default>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_copy_all_artifacts_by_default>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
____________ CopyArtifactsFromFlatDirectoryTest.test_copy_artifacts ____________

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_copy_artifacts>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_copy_artifacts>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_ CopyArtifactsFromFlatDirectoryTest.test_do_nothing_when_not_copying_or_moving _

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_do_nothing_when_not_copying_or_moving>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_do_nothing_when_not_copying_or_moving>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
__________ CopyArtifactsFromFlatDirectoryTest.test_ignore_media_files __________

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_ignore_media_files>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_ignore_media_files>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
____________ CopyArtifactsFromFlatDirectoryTest.test_move_artifacts ____________

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_move_artifacts>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_move_artifacts>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_ CopyArtifactsFromFlatDirectoryTest.test_only_copy_artifacts_matching_configured_extension _

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_only_copy_artifacts_matching_configured_extension>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_only_copy_artifacts_matching_configured_extension>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_ CopyArtifactsFromFlatDirectoryTest.test_prune_import_directory_when_emptied __

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_prune_import_directory_when_emptied>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_prune_import_directory_when_emptied>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_________ CopyArtifactsFromFlatDirectoryTest.test_rename_when_copying __________

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_rename_when_copying>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_rename_when_copying>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
__________ CopyArtifactsFromFlatDirectoryTest.test_rename_when_moving __________

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_rename_when_moving>

    def setUp(self):
>       super(CopyArtifactsFromFlatDirectoryTest, self).setUp()

tests/test_flatdirectory.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_flatdirectory.CopyArtifactsFromFlatDirectoryTest testMethod=test_rename_when_moving>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
______ CopyArtifactsPrintIgnoredTest.test_do_not_print_ignored_by_default ______

self = <tests.test_printignored.CopyArtifactsPrintIgnoredTest testMethod=test_do_not_print_ignored_by_default>

    def setUp(self):
>       super(CopyArtifactsPrintIgnoredTest, self).setUp()

tests/test_printignored.py:12: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_printignored.CopyArtifactsPrintIgnoredTest testMethod=test_do_not_print_ignored_by_default>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_______________ CopyArtifactsPrintIgnoredTest.test_print_ignored _______________

self = <tests.test_printignored.CopyArtifactsPrintIgnoredTest testMethod=test_print_ignored>

    def setUp(self):
>       super(CopyArtifactsPrintIgnoredTest, self).setUp()

tests/test_printignored.py:12: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_printignored.CopyArtifactsPrintIgnoredTest testMethod=test_print_ignored>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_ CopyArtifactsReimportTest.test_do_nothing_when_paths_do_not_change_with_copy_import _

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_do_nothing_when_paths_do_not_change_with_copy_import>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_do_nothing_when_paths_do_not_change_with_copy_import>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_ CopyArtifactsReimportTest.test_do_nothing_when_paths_do_not_change_with_move_import _

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_do_nothing_when_paths_do_not_change_with_move_import>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_do_nothing_when_paths_do_not_change_with_move_import>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
___ CopyArtifactsReimportTest.test_prune_empty_directories_with_copy_import ____

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_prune_empty_directories_with_copy_import>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_prune_empty_directories_with_copy_import>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_________ CopyArtifactsReimportTest.test_reimport_artifacts_with_copy __________

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_reimport_artifacts_with_copy>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_reimport_artifacts_with_copy>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
_________ CopyArtifactsReimportTest.test_reimport_artifacts_with_move __________

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_reimport_artifacts_with_move>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_reimport_artifacts_with_move>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
________ CopyArtifactsReimportTest.test_rename_when_paths_do_not_change ________

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_rename_when_paths_do_not_change>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_rename_when_paths_do_not_change>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
____________ CopyArtifactsReimportTest.test_rename_with_copy_import ____________

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_rename_with_copy_import>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_rename_with_copy_import>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
____________ CopyArtifactsReimportTest.test_rename_with_move_import ____________

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_rename_with_move_import>

    def setUp(self):
        """
        Setup an import directory of the following structure:
    
            testlib_dir/
                Tag Artist/
                    Tag Album/
                        Tag Title 1.mp3
                        artifact.file
        """
>       super(CopyArtifactsReimportTest, self).setUp()

tests/test_reimport.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/helper.py:59: in setUp
    super(CopyArtifactsTestCase, self).setUp()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_reimport.CopyArtifactsReimportTest testMethod=test_rename_with_move_import>

    def setUp(self):
        # A "clean" source list including only the defaults.
        beets.config.sources = []
        beets.config.read(user=False, defaults=True)
    
        # Direct paths to a temporary directory. Tests can also use this
        # temporary directory.
        self.temp_dir = util.bytestring_path(tempfile.mkdtemp())
    
        beets.config['statefile'] = \
>           util.py3_path(os.path.join(self.temp_dir, b'state.pickle'))
E       AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you mean: 'syspath'?

tests/_common.py:55: AttributeError
=============================== warnings summary ===============================
../../nix/store/c9w0rdirjhqmcdlfnip0wcad0hydysvp-python3.12-confuse-1.7.0/lib/python3.12/site-packages/confuse/util.py:126
  /nix/store/c9w0rdirjhqmcdlfnip0wcad0hydysvp-python3.12-confuse-1.7.0/lib/python3.12/site-packages/confuse/util.py:126: DeprecationWarning: 'pkgutil.get_loader' is deprecated and slated for removal in Python 3.14; use importlib.util.find_spec() instead
    loader = pkgutil.get_loader(name)

../../nix/store/03vkxjwpw81kxdf3znpghkwb8dlrlh4c-python3.12-mediafile-0.12.0/lib/python3.12/site-packages/mediafile.py:52
  /nix/store/03vkxjwpw81kxdf3znpghkwb8dlrlh4c-python3.12-mediafile-0.12.0/lib/python3.12/site-packages/mediafile.py:52: DeprecationWarning: 'imghdr' is deprecated and slated for removal in Python 3.13
    import imghdr

tests/helper.py:13
  /build/source/tests/helper.py:13: UserWarning: beets.mediafile is deprecated; use mediafile instead
    from beets import mediafile

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_filename.py::CopyArtifactsFilename::test_import_dir_with_illegal_character_in_album_name - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_filename.py::CopyArtifactsFilename::test_import_dir_with_unicode_character_in_artifact_name_copy - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_filename.py::CopyArtifactsFilename::test_import_dir_with_unicode_character_in_artifact_name_move - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_copy_all_artifacts_by_default - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_copy_artifacts - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_do_nothing_when_not_copying_or_moving - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_ignore_media_files - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_move_artifacts - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_only_copy_artifacts_matching_configured_extension - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_prune_import_directory_when_emptied - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_rename_when_copying - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_flatdirectory.py::CopyArtifactsFromFlatDirectoryTest::test_rename_when_moving - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_printignored.py::CopyArtifactsPrintIgnoredTest::test_do_not_print_ignored_by_default - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_printignored.py::CopyArtifactsPrintIgnoredTest::test_print_ignored - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_do_nothing_when_paths_do_not_change_with_copy_import - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_do_nothing_when_paths_do_not_change_with_move_import - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_prune_empty_directories_with_copy_import - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_reimport_artifacts_with_copy - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_reimport_artifacts_with_move - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_rename_when_paths_do_not_change - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_rename_with_copy_import - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
FAILED tests/test_reimport.py::CopyArtifactsReimportTest::test_rename_with_move_import - AttributeError: module 'beets.util' has no attribute 'py3_path'. Did you me...
SKIPPED [1] tests/test_reimport.py:152: Todo
SKIPPED [1] tests/test_reimport.py:77: Failing
================== 22 failed, 2 skipped, 3 warnings in 0.40s ===================
```

</details>

astratagem committed:
beetsPackages.copyartifacts: beets 2.1.0 compat

montchr commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358086`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>beets</li>
    <li>beets-unstable</li>
    <li>beets-unstable.dist</li>
    <li>beets-unstable.doc</li>
    <li>beets-unstable.man</li>
    <li>beets.dist</li>
    <li>beets.doc</li>
    <li>beets.man</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
  </ul>
</details>

montchr commented:
I've been using `beets-stable` from this PR in my configuration and it's working as expected:

https://github.com/montchr/dotfield/commit/f7391d42e89e11667afc0ffb9b095c7b052db556

Up until the release of 2.1.0, I had been applying a patch via `beets-unstable.override` to work around https://github.com/beetbox/beets/issues/5473, so I was happy to see the new release.

doronbehar commented on pkgs/tools/audio/beets/common.nix:
```diff
@@ -39,15 +39,7 @@ let
   , enable ? !disableAllPlugins
   , builtin ? false
   , propagatedBuildInputs ? [ ]
-  , testPaths ? [
-    # NOTE: This conditional can be removed when beets-stable is updated and
-    # the default plugins test path is changed
-    (if (lib.versions.majorMinor version) == "1.6" then
-      "test/test_${name}.py"
-    else
-      "test/plugins/test_${name}.py"
-    )
-  ]
```
This change can be detached from the update commit.

doronbehar commented on pkgs/tools/audio/beets/common.nix:
```diff
@@ -68,26 +60,31 @@ let
 
   pluginWrapperBins = concatMap (p: p.wrapperBins) (attrValues enabledPlugins);
+
+  disabledTestPaths = lib.flatten (attrValues (lib.mapAttrs (_: v: v.testPaths) disabledPlugins));
```
This change too is very welcome, but can be detached from the update commit.

doronbehar commented on pkgs/tools/audio/beets/default.nix:
```diff
@@ -21,14 +21,12 @@ lib.makeExtensible (self: {
 
   beets-stable = callPackage ./common.nix rec {
     inherit python3Packages;
-    # NOTE: ./builtin-plugins.nix and ./common.nix can have some conditionals
-    # be removed when stable version updates
```
I think that this change can too be detached from the update commit.

doronbehar reviewed (commented):
Looks pretty good to me! Thanks a lot, I wonder why only now I got pinged for a review. Ideally, I'd have liked to see the changes in a different order. The following 4, are fixes / improvements to the current state of the package (many thanks for that!):

```
beets: add @montchr as co-maintainer
beetsPackages.extrafiles: remove broken/unmaintained plugin
beets: mark builtin acousticbrainz plugin as deprecated
beets: alphabetize builtin plugins
beets: small reformatting and dead code removal
```

And the rest of the PR is really only an update to `beets-{,un}stable`, so can be squashed to 2 commit:

```
beets-stable: 2.0.0 -> 2.1.0
beets-unstable: 2.0.0-unstable-2024-05-25 -> 2.1.0-unstable-2024-11-22
```

montchr commented:
@doronbehar Thanks for the review!  I'm working on addressing those changes.  

> And the rest of the PR is really only an update to `beets-{,un}stable`, so can be squashed to 2 commit

What about the changes to `beetsPackages.copyartifacts` to make its tests work with 2.1.0?  Since that's technically a separate package, I would think those changes should still follow the commit convention, thus requiring a separate commit beginning with `beetsPackages.copyartifacts `.

doronbehar commented:
Yes these commit conventions should be good for that package too, as this way ofborg will automatically be triggered to build that plugin as well.

doronbehar committed:
beets: nixfmt & remove dead code

doronbehar committed:
beets: 2.0.0 -> 2.2.0

doronbehar committed:
beets-unstable: 2.0.0-unstable-2024-05-25 -> 2.2.0-unstable-2024-12-02

doronbehar commented:
I fixed the commit log as I wanted, and fixed the merge conflict (due to https://github.com/NixOS/nixpkgs/pull/357903). Now this should be good after CI is green.

doronbehar commented:
Ofborg failed to eval here, but the new evaluations succeeded. I also tested manually the build on aarch64-darwin, so I'm merging.
